src/main/webapp/app/pages/userPage/UserPage.tsx

```diff
@@ -412,13 +412,30 @@ export default class UserPage extends React.Component<IUserPage> {
         ...this.user.additionalInfo,
         userCompany: updatedUserCompany,
       };
-      if (!values.authorities.includes(AUTHORITIES.API)) {
+
+      // If previously the user has API role, but in the latest, it's removed, then we will clean up the api access request
+      if (
+        this.user.authorities.includes(AUTHORITIES.API) &&
+        !values.authorities.includes(AUTHORITIES.API)
+      ) {
         updatedAdditionalInfo.apiAccessRequest = {
           requested: false,
           justification: '',
         };
       }
 
+      if (values.additionalInfoApiAccessJustification) {
```

Comment:
So if an admin modifies the justification, and API request will automatically be created? Could this be an issue if we want to hold onto their request justification, but an admin modifies it, generating a new request?